### ISSUE TITLE
Search.json 500 error (solr 400 error)

### ISSUE THREAD

**Issue by jdlrobson** [internetarchive/openlibrary#1850]
> What problem are we solving? What does the experience look like today? What are the symptoms?

I could have sworn we talked about this in Slack, but couldn't find an open bug - when I run a search for "everything" using the search input at the top of the page, no search results render and I see the search query got a 500 error.

http://localhost:8080/search.json?q=Tales%20from&limit=10&mode=everything&debug=true

Under the hood in the logs I see that a query is being made to solr with following URI:
```
http://solr:8080/solr/select?fq=type:work&q.op=AND&start=0&rows=10&fl=*&defType=dismax&qf=text+title^5+author_name^5&bf=sqrt(edition_count)^10&q=Tales%20from&spellcheck=true&spellcheck.count=3&facet=true&facet.field=has_fulltext&facet.field=author_facet&facet.field=language&facet.field=first_publish_year&facet.field=publisher_facet&facet.field=subject_facet&facet.field=person_facet&facet.field=place_facet&facet.field=time_facet&facet.field=public_scan_b&wt=json
```

This uri throws a 400 response:

```
web_1        | 2019-01-26 21:30:19 [openlibrary.worksearch] [ERROR] Failed solr query
web_1        | Traceback (most recent call last):
web_1        |   File "/openlibrary/openlibrary/plugins/worksearch/code.py", line 179, in execute_solr_query
web_1        |     solr_result = urllib2.urlopen(url, timeout=3)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
web_1        |     return opener.open(url, data, timeout)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 435, in open
web_1        |     response = meth(req, response)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
web_1        |     'http', request, response, code, msg, hdrs)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 473, in error
web_1        |     return self._call_chain(*args)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
web_1        |     result = func(*args)
web_1        |   File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
web_1        |     raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
web_1        | HTTPError: HTTP Error 400: Bad Request
web_1        | Traceback (most recent call last):
```

**Comment by tfmorris** [comment]
Is this reproducible? Does it happen only in the dev configuration? 

That solr url resolves without a problem against a test solr instance, so I suspect some local configuration issue with your dev solr setup.

**Comment by guyjeangilles** [comment]
I can duplicate this on dev...
![#1850_bug](https://user-images.githubusercontent.com/7256157/68969025-e76af480-07db-11ea-819d-a3f36874f385.png)
but not on production.
![#1850_prod](https://user-images.githubusercontent.com/7256157/68969165-3add4280-07dc-11ea-89b2-d3ea5dfe43dc.png)

Assigning @cdrini per slack discussions since this is docker related.

**Comment by cdrini** [comment]
This was fixed by #3117 ; closing.
